<launch>

<!-- Default folder with maps-->

<arg name="folder" default="/home/fuhai/maps"/>

<!-- Distance Measurement Node -->
<node pkg="stroll_bearnav" type="odometry_monitor_node" name="odometry_monitor">
<param name="maxAngularSpeed" value="5" />
<remap from="/odom" to="/encoder_odom"/>
</node>

<!-- feature extraction -->
  <node pkg="stroll_bearnav" type="feature_extraction_node" name="feature_extraction">
  <!-- <remap from="/image" to="/zed_node/left_raw/image_raw_color"/> -->
  <!-- <remap from="/imager" to="/zed_node/right_raw/image_raw_color"/> -->
  <remap from="/image_event" to="/dvs_rendering"/>
  </node>

<!-- mapping node -->
<node pkg="stroll_bearnav" type="mapper_node" name="mapper" output="screen">
  <!-- <remap from="/image" to="/zed_node/left/image_rect_color"/> -->
  <!-- <remap from="/imager" to="/zed_node/right/image_rect_color"/> -->
  <remap from="/image_event" to="/dvs_rendering"/>
  <param name="folder" value="$(arg folder)" />
  <param name="maxForwardSpeed" type="double" value="5" />
  <param name="maxAngularSpeed" type="double" value="4" />
  <param name="maxForwardAcceleration" type="double" value="0.05" />
</node>

<!-- map loader node -->
<node pkg="stroll_bearnav" type="map_preprocessor_node" name="map_preprocessor" output="screen">
<param name="folder" value="$(arg folder)" />
<param name="image_only" value="true" />
</node>

<!-- navigator node -->
<node pkg="stroll_bearnav" type="end2end_navigator_node" name="navigator" output="screen">
<!-- <remap from="/image_stereo" to="/zed_node/stereo/image_rect_color"/> -->
<remap from="/image_event" to="/dvs_rendering"/>
<!-- <remap from="/image" to="/zed_node/left/image_rect_color"/> -->
  <param name="folder" value="$(arg folder)" />
  <param name="write_log" type="bool" value="true"/>
  <param name="PID_Kp" type="double" value="0.4"/> 
  <param name="PID_Ki" type="double" value="0.1"/>
  <param name="PID_Kd" type="double" value="0.3"/>
</node>


<!-- Run robot driver, teleop and sensors -->
<include file="$(find jaguar4x4_2014)/launch/move_and_sense.launch" />


<!-- Launch stroll GUI -->
<include file="$(find stroll_bearnav)/launch/stroll-gui.launch" />

<!-- rviz for sensors vizualization -->
<node pkg="rviz" type="rviz" name="sensors_rviz" args="-d $(find stroll_bearnav)/rviz/sensors.rviz" />

</launch>
